lightning-js/ui-components
# fix: rollback eslint

## Description

Turns out linting was broken because we updated to v9 which makes a bunch of API changes. Rolling us back to 8 since eslint/typescript hasn't been updated to support 9 yet

## Changes
- roll eslint back to v8
- stop linting theme for the time being
- make unused vars rule more permissive
- resolve a handful of missed linting issues

## Testing
- run the linter and ensure it's running correctly instead of erroring
- run a build, ensure stories still render right(this doesn't touch any real logic so this should all be fine)
